app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -622,10 +771,27 @@ public void onClick(DialogInterface dialog, int which) {
     private void setupActionBarHandler(final StreamInfo info) {
         if (DEBUG) Log.d(TAG, "setupActionBarHandler() called with: info = [" + info + "]");
         sortedStreamVideosList = new ArrayList<>(ListHelper.getSortedStreamVideosList(activity, info.getVideoStreams(), info.getVideoOnlyStreams(), false));
+        actionBarHandler.setStreamResolution(oldSelectedStreamResolution);
         actionBarHandler.setupStreamList(sortedStreamVideosList, spinnerToolbar);
+
+        actionBarHandler.setOnStreamSelectedListener(new ActionBarHandler.OnStreamChangesListener() {
+            @Override
+            public void onActionSelected (int selectedStreamId) {
+                if(mPlayerService != null && oldSelectedStreamId != selectedStreamId && selectedStreamId != -1) {
```

Comment:
nitpicking, but we should have a space between "if" and parentheses like all the existing code.

Reply:
More than 100 search results of "if(" in the project. I can replace them all, it's pretty easy.